test/distributed/_tensor/test_math_ops.py

```diff
@@ -200,7 +210,112 @@ def _replicate_fn(name, module, device_mesh):
+        for shard_dim, norm_idx, elementwise_affine in test_config_list:
+            x = torch.rand(
+                batch,
+                sentence_length,
+                embedding_dim,
+                device=self.device_type,
+                requires_grad=True,
+            )
+            normalized_shape = x.shape[norm_idx:]
+            layer_norm = torch.nn.LayerNorm(
+                normalized_shape,
+                elementwise_affine=elementwise_affine,
+                device=self.device_type,
+            )
+            layer_norm_local = copy.deepcopy(layer_norm).to(self.device_type)
+
+            def _replicate_fn(name, module, device_mesh):
+                for name, param in module.named_parameters():
+                    if name in ["weight", "bias"]:
+                        param_dist = torch.nn.Parameter(
+                            distribute_tensor(param, device_mesh, [Replicate()])
+                        )
+                        module.register_parameter(name, param_dist)
+
+            layer_norm_dist = distribute_module(layer_norm, device_mesh, _replicate_fn)
+
+            if elementwise_affine:
+                self.assertEqual(
+                    layer_norm_local.weight, layer_norm_dist.weight.full_tensor()
+                )
+                self.assertEqual(
+                    layer_norm_local.bias, layer_norm_dist.bias.full_tensor()
+                )
+
+            x_local = x.detach().clone().requires_grad_(True)
+            x_dist = distribute_tensor(x, device_mesh, [Shard(shard_dim)])
+            self.assertEqual(x_local, x_dist.full_tensor())
+
+            y_local = layer_norm_local(x_local)
+            # make sure that backward layer norm does not introduce extra collectives
+            comm_mode = CommDebugMode()
+            with comm_mode:
+                y_dist = layer_norm_dist(x_dist)
+
+            self.assertEqual(
+                comm_mode.get_total_counts(),
+                0,
+                f"comm count={comm_mode.get_total_counts()}, "
+                f"shard_dim={shard_dim}, norm_shape={normalized_shape}, elem_affine={elementwise_affine}",
+            )
+
+            self.assertEqual(y_local, y_dist.full_tensor())
+
+            # backward step
+            y_local.sum().backward()
+            with comm_mode:
+                y_dist.sum().backward()
+
+            self.assertEqual(
+                comm_mode.get_total_counts(),
+                0,
+                f"comm count={comm_mode.get_total_counts()}, "
+                f"shard_dim={shard_dim}, norm_shape={normalized_shape}, elem_affine={elementwise_affine}",
+            )
+
+            if elementwise_affine:
+                # if input is sharded on any outer dimension, the gradient of weight
+                # and bias should be _Partial
+                dim_map = x_dist._spec.dim_map
+                outer_dims = range(norm_idx)
+                needs_reduction = any(dim_map[d] >= 0 for d in outer_dims)
```

Comment:
Can we just use `assert isinstance(grad.placements[0], Partial)`. 

The `dim_map` property would not include partial sum informations so the check here seems wrong?